tests/phpunit/tests/link/getAdjacentPost.php

```diff
@@ -350,4 +350,81 @@ public function filter_excluded_terms( $excluded_terms ) {
 		$excluded_terms[] = $this->exclude_term;
 		return $excluded_terms;
 	}
+
+	/**
+	 * @ticket 41131
+	 */
+	public function test_get_adjacent_post_cache() {
+		global $wpdb;
+		// Need some sample posts to test adjacency.
+		$post_one = self::factory()->post->create_and_get(
+			array(
+				'post_title' => 'First',
+				'post_date'  => '2012-01-01 12:00:00',
+			)
+		);
+
+		$post_two = self::factory()->post->create_and_get(
+			array(
+				'post_title' => 'Second',
+				'post_date'  => '2012-02-01 12:00:00',
+			)
+		);
+
+		$post_three = self::factory()->post->create_and_get(
+			array(
+				'post_title' => 'Third',
+				'post_date'  => '2012-03-01 12:00:00',
+			)
+		);
+
+		$post_four = self::factory()->post->create_and_get(
+			array(
+				'post_title' => 'Fourth',
+				'post_date'  => '2012-04-01 12:00:00',
+			)
+		);
+
+		// Assign some terms
```

Comment:
Nitpick: full stops on sentences in comments.

```suggestion
		// Assign some terms.
```

:1234: This applies to multiple comments in this.